MDP-206: Homepage dynamic rendering of articles

Stephaaan opened the pull request.

Commit:
Homepage dynamic rendering of articles

garySiladi commented on src/components/Layout/index.tsx:
```diff
@@ -0,0 +1,13 @@
+import React from "react";
```
I'm not sure if you're still working on this file. The children don't need to be typed, just use the FC type of React.
```
import React, { FC } from "react";
import { Header } from "../Header/Header";
import { Footer } from "../Footer";

export const Layout: FC = ({ children }) => (
  <>
    <Header />
    {children}
    <Footer />
  </>
);
```

Commit:
Fixing homepage

Commit:
Changed maximal width of content on the page

Commit:
Fixed bugs in header component

fixed - closing hamburger menu when user's switching from mobile to desktop version
-hamburger menu render

Commit:
Updated Image component, fixed file name

Image in article thubmnail is now loaded from database instead of showing default Homer image...

Commit:
Merge branch 'master' into sr/Homepage

Commit:
Fixed image style

Commit:
fixed design issues -> need to fix issue w button and check if the page fits to the design

Commit:
...

Commit:
Fixed design issues caused by moving to the newer version of Tailwind CSS

OlehVynnyk commented on src/components/Header/Header.tsx:
```diff
@@ -14,18 +14,22 @@ export const Header = () => {
-          <div className="flex justify-end mr-24 mt-16">
-            <div onClick={() => setOpenMenu(false)}>
-              <ImageHeader src={closeImg} />
+        <>
```
When clicking at the user icon, user profile work incorrect.

OlehVynnyk commented on src/components/Header/Header.tsx:
```diff
@@ -14,18 +14,22 @@ export const Header = () => {
+            <div className="z-50">
+              <HamburgerMenu />
             </div>
           </div>
```
"Arteterapie", "Rozvrh" and "Clanky" has not correctly position

OlehVynnyk commented on src/components/Homepage/Button.tsx:
```diff
@@ -7,11 +7,12 @@ interface IProps {
+    style={{ lineHeight: "1" }}
     className={classNames(
-      "w-full h-40 flex justify-center items-center",
+      "w-full h-40 flex justify-center items-center items-center",
```
Change button font

Commit:
Fixed bugs according to Oleh code review

OlehVynnyk commented on src/components/Header/Header.tsx:
```diff
@@ -14,42 +14,50 @@ export const Header = () => {
-          <div className="flex justify-end mr-24 mt-16">
-            <div onClick={() => setOpenMenu(false)}>
-              <ImageHeader src={closeImg} />
+        <>
```
Rename Header.tsx to index.tsx

OlehVynnyk commented on src/components/Homepage/Button.tsx:
```diff
@@ -7,9 +7,10 @@ interface IProps {
 export const Button: FC<IProps> = ({
   children,
-  className = "text-white font-lato font-bold"
+  className = "text-white font-lato font-bold "
```
delete space at the end of div

OlehVynnyk commented on src/components/Homepage/LandingPageWrapper.tsx:
```diff
@@ -25,10 +25,11 @@ export const LandingPageWrapper = () => {
           Vyskúšajte si CHAT. Registrujte sa a získajte prvú hodinu zdarma.
         </Description>
-        <div className="w-192 m-auto mt-24 bg-black rounded-8">
+        <div className="w-192 m-auto mt-24 bg-black rounded-8 ">
```
same problem with space

Commit:
fixed bugs according to Oleh's review

garySiladi commented on src/components/Homepage/NewestArticles.tsx:
```diff
@@ -0,0 +1,60 @@
+  const [articles, setArticles] = useState([]);
+  const [loadingError, setLoadingError] = useState(false);
+  useEffect(() => {
+    fetch("http://localhost/api/article/newest")
```
don't we have a real endpoint running?

garySiladi replied:
also have the given endpoint (`http://localhost/api/`) from a config file imported

pggl replied:
we need create env file

garySiladi commented on src/components/Homepage/NewestArticles.tsx:
```diff
@@ -0,0 +1,60 @@
+  id: number;
+  heading: string;
+  insight: string;
+  title_photo: string;
```
naming should be `photoTitle`

Stephaaan replied:
this is name that's returned from backend (this is default laravel convention), so, should I leave it like that or add a few lines to code to rename that key?

pggl replied:
rename it

garySiladi commented on src/components/Homepage/NewestArticles.tsx:
```diff
@@ -0,0 +1,60 @@
+          )
+        ) : (
+          articles.map((article: IArticle) => (
+            <div
```
create an article component in a separate file which instances you would return from this .map

garySiladi replied:
that new file should contain the IArticle interface

garySiladi commented on src/components/Homepage/NewestArticles.tsx:
```diff
@@ -0,0 +1,60 @@
+      <h1 className="font-gilbert text-31 mb-16">Najnovšie články</h1>
+      <div className="flex flex-col items-center justify-center lg:flex-row">
+        {articles.length === 0 ? (
+          loadingError ? (
```
don't have nested conditions + defensive programming, so first you solve the exception cases

pggl commented on src/components/Homepage/Button.tsx:
```diff
@@ -10,6 +10,7 @@ export const Button: FC<IProps> = ({
   className = "text-white font-lato font-bold"
 }) => (
   <button
+    style={{ lineHeight: "1" }}
```
use tailwind

pggl commented on src/components/Homepage/NewestArticles.tsx:
```diff
@@ -0,0 +1,60 @@
+                icon={
+                  "http://localhost/api/image/214/160/" + article.title_photo
+                }
+                alt="super"
```
why is "super" hardcoded?

Stephaaan replied:
because we don't have a field for title photo description on backend. Is it necessary?

Stephaaan replied:
Is it okay to add as alt sentence: The title photo of article ${article.title}?

pggl replied:
maybe article name is good for alt

Commit:
fixed issues according to senior review
